### PR TITLE
CodeQL docs: further updates to 'Learn CodeQL' project

### PR DESCRIPTION
**Pull request by jf205**
A few more things that slipped through the cracks when working on the `docs-preparation` branch.

Previews:
- [CodeQL for C and C++](http://docteam.internal.semmle.com/james/tidy-up-branch/learn-ql/cpp/ql-for-cpp.html)--some topics were missed from the toc.
- [CodeQL for Java](http://docteam.internal.semmle.com/james/tidy-up-branch/learn-ql/java/ql-for-java.html)--fixed a couple of typos.

### CODE REVIEW

File: docs/language/learn-ql/cpp/ql-for-cpp.rst [hunk 1]
```diff
@@ -34,6 +34,12 @@ Experiment and learn how to write effective and efficient queries for CodeQL dat
 
 -  :doc:`Detecting a potential buffer overflow <zero-space-terminator>`: You can use CodeQL to detect potential buffer overflows by checking for allocations equal to ``strlen`` in C and C++.
 
+-  :doc:`Using range analysis for C and C++ <range-analysis>`: You can use range analysis to determine the upper or lower bounds on an expression, or whether an expression could potentially over or underflow.
+
+-  :doc:`Hash consing and value numbering <value-numbering-hash-cons>`: You can use specialized CodeQL libraries to recognize expressions that are syntactically identical or compute the same value at runtime in C and C++ codebases.
+
+-  :doc:`Using the guards library in C and C++ <guards>`: You can use the CodeQL guards library to identify conditional expressions that control the execution of other parts of a program in C and C++ codebases.
```
Comment (shati-patel) [inline docs/language/learn-ql/cpp/ql-for-cpp.rst:41]:
```suggestion
-  :doc:`Using the guards library in C and C++ <guards>`: You can use the CodeQL guards library to identify conditional expressions that control the execution of other parts of a program in C and C++ codebases. 

-  :doc:`Using range analysis for C and C++ <range-analysis>`: You can use range analysis to determine the upper or lower bounds on an expression, or whether an expression could potentially over or underflow.

-  :doc:`Hash consing and value numbering <value-numbering-hash-cons>`: You can use specialized CodeQL libraries to recognize expressions that are syntactically identical or compute the same value at runtime in C and C++ codebases.
```

This order matches the side bar ^

File: docs/language/learn-ql/java/ql-for-java.rst [hunk 2]
```diff
@@ -18,7 +18,7 @@ Experiment and learn how to write effective and efficient queries for CodeQL dat
 
 -  `Basic Java query <https://lgtm.com/help/lgtm/console/ql-java-basic-example>`__: Learn to write and run a simple CodeQL query using LGTM.
 
--  :doc:`CodeQL library for Java <introduce-libraries-java>`: When analyzing C or C++ code, you can use the large collection of classes in the CodeQL library for C and C++.
+-  :doc:`CodeQL library for Java <introduce-libraries-java>`: When analyzing C or C++ code, you can use the large collection of classes in the CodeQL library for Java.
```
Comment (shati-patel) [inline docs/language/learn-ql/java/ql-for-java.rst:21]:
```suggestion
-  :doc:`CodeQL library for Java <introduce-libraries-java>`: When analyzing Java code, you can use the large collection of classes in the CodeQL library for Java.
```

**Comment by shati-patel** [comment]
(There also seems to be an extra space after "Basic Python query" in the [overview topic](http://docteam.internal.semmle.com/james/tidy-up-branch/learn-ql/python/ql-for-python.html), but that's very much a minor thing 😉)

**Comment by jf205** [comment]
Thanks @shati-patel! Luckily we have time to sort this out now 👍

**Pull request merged**